docs/ldml/tr35-keyboards.md

```diff
@@ -308,6 +308,96 @@ For extensibility, the `<special>` element will be allowed at nearly every level
+
+The 'gluing' is only applicable during one particular processing step. It does not persist or affect further processing steps or future keystrokes.
+
+A second example:
+
+**Example 2**
+
+- `e\m{marker0}\u{0300}\m{marker1}\u{0320}\m{marker2}` (original)
+- `e\m{marker1}\u{0320}\m{marker0}\u{0300}\m{marker2}` (NFD)
+
+Here `\m{marker2}` is 'glued' to the end of the segment. However, if additional text is added such as by a subsequent keystroke (which may add an additional combining character, for example), this marker may be 'glued' to that following text.
+
+Markers remain in the same normalization-safe segment during normalization. Consider:
+
+**Example 3**
+
+- `e\u{0300}\m{marker1}\u{0320}a\u{0300}\m{marker2}\u{0320}` (original)
+- `e\m{marker1}\u{0320}\u{0300}a\m{marker2}\u{0320}\u{0300}` (NFD)
+
+There are two normalization-safe segments here:
+
+1. `e\u{0300}\m{marker1}\u{0320}`
+2. `a\u{0300}\m{marker2}\u{0320}`
+
+Normalization (and marker rearranging) occurs within each segment.  While `\m{marker1}` is 'glued' to the `\u{0320}`, it is glued within the first segment and has no effect on the second segment.
+
+#### Rationale for 'gluing' markers
+
+It is recognized that the processing described here seems to be an innovation among Unicode normalization implementations.
+
+This specification has markers 'glued' (remaining with) the following character so that if a context ends with a marker, that marker would be guaranteed to remain at the end after processing.  Authors can keep a marker together with a character of interest by emitting the marker just before the character of interest, that is, `output="\m{marker}X"` instead of `output="X\m{marker}"`.
+
+### Normalization and Character Classes
+
+If pre-composed (non-NFD) characters are used in [character classes](#regex-like-syntax), such as `[á-é]` or `[\u{00e1}-\u{00e9}]`, these may not match as keyboard authors expect, as the U+00E1 character will not occur in NFD form.
+
+The above could be written instead as a  `(á|â|ã|ä|å|æ|ç|è|é)`, or as a set variable `<set id="Ex" value="á â ã ä å æ ç è é"/>` and matched as `$[Ex]`.
+
+Implementations may want to warn users when character classes include non-NFD characters.
+
+### Normalization and Output
+
+On output, text will be normalized into the form requested by that implementation, or possibly specifically requested by a particular application.
+For example, many platforms may request NFC as the output format. In such a case, all text emitted via the keyboard will be transformed into NFC.
+
+Existing text in a document will only have normalization applied within a single normalization-safe segment from the caret.
```

Comment:
```suggestion
Existing text in a document will only have normalization applied within a single normalization-safe segment from the caret.  Output normalization to NFC, when appropriate, is unaffected by any markers embedded within the segment.
```

Just to be extra explicit about something that's currently just implied.

Reply:
Unclear what is meant. If I have a<marker>\u0308 then the a + \u0308 converts to ä under NFC. In that circumstance is the marker to be removed or moved?

Reply:
Output doesn't include markers.  

Just making clear that having the marker doesn't prevent character composition when there _are_ markers between the decomposed pieces within the LDML keyboard "black box".

Reply:
I was thinking that, and so wondering about the language. Might be good to include that example for clarity after your sentence:

For example, U`marker`\u0308 is output as Ü.

Reply:
good idea. I'll add an example.